### DISCUSSION TITLE
Graphql mutation file upload with file data in single query

### DISCUSSION THREAD

**Discussion by NXC-Controls** [strapi/strapi#6660] [Legacy Topics]
i have created a student form of two filed "name"(text) & "picture"(media) . i am using graphql as api end point .
i can run mutation query of file upload alone with
`{
  query: `
    mutation($file: Upload!) {
      singleUpload(file: $file) {
        id
      }
    }
  `,
  variables: {
    file: File // image
  }
}` 
i
But i want to do mutation of createStudent with name & file in single query.

**Comment by robclouth** [comment]
Did you figure this out?

**Reply by NXC-Controls** [reply]
No , we are uploading first file then send field parameters separately . if you find any solution then Please do reply .

**Reply by robclouth** [reply]
I'll just do that then for now. Thanks!
